app/code/Magento/MediaGalleryUi/view/adminhtml/web/js/action/deleteImageWithDetailConfirmation.js

```diff
@@ -68,7 +70,26 @@ define([
                 return '';
             }
 
-            return usedInMessage + usedIn.join(', ') + '.';
+            return usedInMessage + this.usedInObjectToString(usedIn);
+        },
+
+        /**
+         * Fromats usedIn object to string
+         *
+         * @param {Object} usedIn
+         * @return {String}
+         */
+        usedInObjectToString: function (usedIn) {
+            var message = '',
+                count = 0;
+
+            $.each(usedIn, function (entityName, number) {
+                count++;
+                message += entityName + '(' + number + ')';
+                message += count !== Object.keys(usedIn).length ?  ', ' : '.';
```

Comment:
I think it would be better readable if the object will be converted to an array and then rendered using `.join(', ') + '.'`